lib/src/events/guild_events.dart

```diff
@@ -411,3 +412,42 @@ class GuildStickerUpdate implements IGuildStickerUpdate {
     stickers = [for (final rawSticker in raw["d"]["stickers"]) GuildSticker(rawSticker as RawApiMap, client)];
   }
 }
+
+abstract class IGuildEventCreateEvent {
+  IGuildEvent get event;
+}
+
+class GuildEventCreateEvent implements IGuildEventCreateEvent {
+  @override
+  late final IGuildEvent event;
+
+  GuildEventCreateEvent(RawApiMap raw, INyxx client) {
+    event = GuildEvent(raw['d'] as RawApiMap, client);
+  }
+}
+
+abstract class IGuildEventUpdateEvent {
+  IGuildEvent get event;
+}
+
+class GuildEventUpdateEvent implements IGuildEventUpdateEvent {
+  @override
+  late final IGuildEvent event;
+
+  GuildEventUpdateEvent(RawApiMap raw, INyxx client) {
+    event = GuildEvent(raw['d'] as RawApiMap, client);
+  }
+}
+
+abstract class IGuildEventDeleteEvent {
+  IGuildEvent get event;
+}
```

Comment:
all the events are looking the same, maybe we should just provide one call for all of them? Or do inheritance maybe in case they extends event data in future?

Reply:
gonna leave as is for now :)